src/lib.rs

```diff
@@ -32,9 +32,42 @@
 mod windows;
 
 use std::env;
+use std::ffi::OsString;
 use std::io;
 use std::path::{Path, PathBuf};
 
+/// Permits parameterising the home functions via the _from variants - used for
+/// in-process unit testing by rustup.
+pub trait Env {
```

Comment:
```suggestion
pub trait CustomEnv {
```
or 
```suggestion
pub trait ThreadEnv {
```

may convey more meaning to users.

Reply:
mmm, ThreadEnv would be semantically wrong; TLS binding is just one way to use this; context lookups are the core mechanic, of which TLS bindings are a very common very user friendly form. I don't really follow the 'custom' implication either - all traits are extension points, otherwise we wouldn't need traits, so ...

Reply:
Oh right.